File: LeetCode-Solved-Problems/2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        int n=s.size();
        string ans="";
        stack<char>stck;
        for(int i=0;i<n;i++){
            if(s[i]=='*' and !stck.empty())stck.pop();
            else stck.push(s[i]);
        }
        
        while(!stck.empty()){
            ans+=stck.top();
            stck.pop();
        }
        
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: LeetCode-Solved-Problems/2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    const stack=[];
    for(var i=0;i<s.length;i++){
        if(s[i]=='*' && stack.length){
            stack.pop();
        }
        else{
            stack.push(s[i]);
        }
    }
    
    return stack.join('');
};